var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -84,7 +84,8 @@ def check_compilers(self):
                 'OpenBLAS requires both C and Fortran compilers!'
             )
         # Add support for OpenMP
-        if '+openmp' in self.spec and self.spec.satisfies('%clang'):
+        if (('+openmp' in self.spec) and self.spec.satisfies('%clang') and
+            (('@:0.2.19' in self.spec) or ('platform=darwin' in self.spec))):
```

Comment:
Technically, I think people can build `llvm` on macOS and it should support OpenMP just fine. A better check, which `qt` uses, would be:
```python
str(self.spec.compiler.version).endswith('-apple')
```
You should also update the comment below.